lua/tint.lua

```diff
@@ -169,12 +176,79 @@ local function setup_namespaces()
     __.tint_ns = vim.api.nvim_create_namespace("_tint_dim")
   end
 
-  for hl_group_name, hl_def in pairs(get_global_highlights()) do
+  for hl_group_name, hl_def in pairs(get_highlights(0)) do
     -- Ensure we only have valid keys copied over
     hl_def = ensure_valid_hl_keys(hl_def)
     set_default_ns(hl_group_name, hl_def)
-    set_tint_ns(hl_group_name, hl_def)
+    set_tint_ns(hl_group_name, hl_def, __.tint_ns)
+  end
+end
+
+---@param hl_ns_id number
+---@param suffix string
+local function add_namespace(hl_ns_id, suffix)
+  local ns_name = "tint_ns_" .. tostring(suffix)
+  local ns_id = vim.api.nvim_create_namespace("_tint_dim_" .. tostring(suffix))
+  __[ns_name] = ns_id
```

Comment:
_nit_: We could do something like `__.extra_namespaces` just to make sure things are organized in our `__` map.

Reply:
see above, i like the generalisation like `tint_ns_<id>` - let me know if you prefer otherwise